### PR TITLE
Metall: add v0.24 and v0.25

### PR DESCRIPTION
**Pull request by KIwabuchi**

### CODE REVIEW

**Comment by KIwabuchi** [comment]
Hi Spack team,

Could you also check if the current package configuration is correct for enabling a build-time test?

I thought I was able to run Metall's unit test (`make test`) by typing 'spack install --test={root/all} metall'.
But I'm not sure if it works correctly now.

File: var/spack/repos/builtin/packages/metall/package.py [hunk 1]
```diff
@@ -39,7 +41,7 @@ class Metall(CMakePackage):
     version("0.9", sha256="2d7bd9ea2f1e04136050f210884445a9e3dcb96c992cf42ff9ea4b392f85f927")
 
     depends_on("cmake@3.10:", type="build")
-    depends_on("boost@1.64:", type=("build", "link"))
+    depends_on("boost@1.75:", type=("build", "link"))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/metall/package.py:44]: Does this change apply to all of the listed versions or for some version on?  If the latter, then you could keep the original `depends_on` directive and add a new one with a `when` argument (https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies), e.g., if it applies to `0.24` on then the dependencies could be:

```suggestion
    depends_on("boost@1.64:", type=("build", "link"))
    depends_on("boost@1.75:", when="@0.24:", type=("build", "link"))
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the two version sha256.  Just have a question about the dependency change.

File: var/spack/repos/builtin/packages/metall/package.py (hunk 1 above)
Comment (KIwabuchi) [reply var/spack/repos/builtin/packages/metall/package.py:44]: Yes, the change applies to all of the versions. We are thinking of stopping officially supporting any versions less than 1.75 for Boost.

**Comment by tldahlgren** [comment]
> Hi Spack team,
> 
> Could you also check if the current package configuration is correct for enabling a build-time test?
> 
> I thought I was able to run Metall's unit test (`make test`) by typing 'spack install --test={root/all} metall'. But I'm not sure if it works correctly now.

Have you tried `spack install --verbose --test=root metall`?  When I do that, I see the following, which means your `make check` is being run after the `build` and before the `install` phase as expected from the docs (https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-installation-phase-tests).  

```
==> Testing package metall-0.25-zx7hf2c
==> [2023-04-21-10:32:57.153655] Running build-time tests
==> [2023-04-21-10:32:57.153819] RUN-TESTS: build-time tests [check]
==> metall: Executing phase: 'install'
...
```

You can also look in the install prefix of the package under `.spack/install-time-test-log.txt` to see the test results.  In my case, I'm seeing:

```
==> Testing package metall-0.25-7lavqu3
==> [2023-04-21-10:40:50.929501] Running build-time tests
==> [2023-04-21-10:40:50.929700] RUN-TESTS: build-time tests [check]
...
172/174 Test  #67: ManagerTest.AllSmallAllocation ...........................   Passed    4.01 sec
173/174 Test #155: JSONValueTest.MoveDifferentMetallAllocator ...............   Passed    2.21 sec
174/174 Test  #72: ManagerTest.AlignedAllocation ............................   Passed    4.94 sec

100% tests passed, 0 tests failed out of 174

Total Test time (real) =   6.25 sec
...
```

**Review by tldahlgren — APPROVED** [review]

**Comment by KIwabuchi** [comment]
@tldahlgren Thank you for your help! I checked the 'install-time-test-log.txt' file and confirmed that Metall's test ran as expected.

**Pull request merged**